libraries/ml/src/spe.ts

```diff
@@ -84,9 +89,9 @@ export class SPEBase {
 
     let lambda2 = this.lambda2;
 
-    if (this.steps == 0) {
-      this.steps = vectors.length-1;
-    }
+    if (this.steps == 0)
```

Comment:
strict comparison

Reply:
What's wrong with it though? We know that this.steps is a number, and we'd get a warning if we try to compare incompatible types using "=="

Reply:
why do we even need to use '==' in JS?
strict comparison is safer and faster (doesn't use type casting)